alicat/driver.py

```diff
@@ -0,0 +1,484 @@
+
+    async def __aenter__(self, *args):
+        """Provide async enter to context manager."""
+        return self
+
+    async def __aexit__(self, *args):
+        """Provide async exit to context manager."""
+        return
+
+    @classmethod
+    async def is_connected(cls, port, unit='A') -> bool:
+        """Return True if the specified port is connected to this device.
+
+        This class can be used to automatically identify ports with connected
+        Alicats. Iterate through all connected interfaces, and use this to
+        test. Ports that come back True should be valid addresses.
+
+        Note that this distinguishes between `FlowController` and `FlowMeter`.
+        """
+        is_device = False
+        try:
+            device = cls(port, unit)
+            try:
+                c = await device.get()
+                if cls.__name__ == 'FlowMeter':
+                    assert c
+                    assert 'setpoint' not in device.keys
+                elif cls.__name__ == 'FlowController':
+                    assert c
+                    assert 'setpoint' in device.keys
+                else:
+                    raise NotImplementedError('Must be meter or controller.')
+                is_device = True
+            finally:
+                await device.close()
+        except Exception:
+            pass
+        return is_device
+
+    def _test_controller_open(self) -> None:
+        """Raise an IOError if the FlowMeter has been closed.
+
+        Does nothing if the meter is open and good for read/write
+        otherwise raises an IOError. This only checks if the meter
+        has been closed by the FlowMeter.close method.
+        """
+        if not self.open:
+            raise OSError(f"The FlowMeter with unit ID {self.unit} and "
+                           "port {self.hw.address} is not open")
+
+    def _is_float(self, msg):
+        try:
+            float(msg)
+            return True
+        except ValueError:
+            return False
+
+    async def _write_and_read(self, command):
+        """Wrap the communicator request, to call _test_controller_open() before any request
+        """
+        self._test_controller_open()
+        return await self.hw._write_and_read(command)
+
+    async def get(self) -> dict:
+        """Get the current state of the flow controller.
+
+        From the Alicat mass flow controller documentation, this data is:
+         * Pressure (normally in psia)
+         * Temperature (normally in C)
+         * Volumetric flow (in units specified at time of order)
+         * Mass flow (in units specified at time of order)
+         * Total flow (only on models with the optional totalizer function)
+         * Currently selected gas
+
+        Args:
+            retries: Number of times to re-attempt reading. Default 2.
+        Returns:
+            The state of the flow controller, as a dictionary.
+
+        """
+        command = f'{self.unit}'
```

Comment:
This line's not doing anything, right?

Reply:
<img width="495" alt="Screenshot 2023-04-12 at 12 30 22 PM" src="https://user-images.githubusercontent.com/52292902/231537427-7d150151-ef10-4b62-9b74-e6ef2b05abef.png">